rpc/src/v1/impls/eth_filter.rs

```diff
@@ -176,6 +213,10 @@ impl<T: Filterable + Send + Sync + 'static> EthFilter for T {
 					filter.from_block = BlockId::Number(*block_number);
 					filter.to_block = BlockId::Latest;
 
+					// retrieve reorg logs
+					let (mut reorg, reorg_len) = last_block_hash.map_or_else(|| (Vec::new(), 0), |h| self.canon_logs(h, &filter));
+					*block_number -= reorg_len as u64;
```

Comment:
This doesn't affect the filter that's used below when getting pending logs. And it's overridden afterwards with `current + 1`.

Reply:
My mistake. Filter `from_block` should be set after we alter `block_number`.